p2pd/main.go

```diff
@@ -85,118 +90,223 @@ func main() {
 	announceAddrs := flag.String("announceAddrs", "", "comma separated list of multiaddrs the host should announce to the network")
 	noListen := flag.Bool("noListenAddrs", false, "sets the host to listen on no addresses")
 	metricsAddr := flag.String("metricsAddr", "", "an address to bind the metrics handler to")
+	configFilename := flag.String("f", "", "a file from which to read a json representation of the deamon config")
+	configStdin := flag.Bool("i", false, "have the daemon read the json config from stdin")
 	pprof := flag.Bool("pprof", false, "Enables the HTTP pprof handler, listening on the first port "+
 		"available in the range [6060-7800], or on the user-provided port via -pprofPort")
 	pprofPort := flag.Uint("pprofPort", 0, "Binds the HTTP pprof handler to a specific port; "+
 		"has no effect unless the pprof option is enabled")
 
 	flag.Parse()
 
-	if *pprof {
-		// an invalid port number will fail within the function.
-		go pprofHTTP(int(*pprofPort))
-	}
-
+	var c config.Config
 	var opts []libp2p.Option
 
-	maddr, err := multiaddr.NewMultiaddr(*maddrString)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	if *id != "" {
-		key, err := p2pd.ReadIdentity(*id)
+	if *configStdin {
+		stdin := bufio.NewReader(os.Stdin)
+		body, err := ioutil.ReadAll(stdin)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if err := json.Unmarshal(body, c); err != nil {
+			log.Fatal(err)
+		}
+	} else if *configFilename != "" {
+		body, err := ioutil.ReadFile(*configFilename)
 		if err != nil {
 			log.Fatal(err)
 		}
+		if err := json.Unmarshal(body, c); err != nil {
+			log.Fatal(err)
+		}
+	} else {
```

Comment:
we need to allow command line options to override the config.

Reply:
I think we should do it here, it's a pretty simple thing to do. It would be very surprising to pass options that are ignored when also passing a config.

Reply:
sounds good let’s do overrides